editor.planx.uk/src/@planx/components/Send/uniform/UniformPayload/model.ts

```diff
@@ -52,29 +52,32 @@ export class UniformPayload implements IUniformPayload {
 
     this.proposalCompletionDate = this.setProposalCompletionDate();
     this.siteAddress = passport.data?.["_address"];
+    this.payload = this.mapPassportToUniformPayload();
+  }
 
-    this["portaloneapp:Proposal"] = {
-      ...this.getNamespaces(),
-      _Version: "1.3",
-      "portaloneapp:SchemaVersion": 1.3,
+  private stringToBool = (value: string): boolean | undefined => {
+    if (value) return value.toLowerCase() === "true";
+  };
```

Comment:
I think what I'd ideally like to put in place is a `Passport` class that just had a getter than handles this sort of thing, e.g. - 

```ts
// Passport.ts

class Passport {
  userPassport: Store.passport

  constructor(userPassport: Store.passport) {
    this.userPassport = userPassport
  }


  get = (value: string): boolean | string[] | string | number => {
    // Logic here to correctly access passport variables
  }
}
```

```ts
// model.ts
const passport = new Passport(userPassport)
...
"common:HaveSoughtAdvice": passport.get("application.preAppAdvice")
```

There's enough change in this PR as is, so this is just a helper function for now, but comments on the above most welcome.

Reply:
I think a nice thing about this approach also is that if we only expose a `get()` interface we can more easily change the structure of the passport in future without breaking too much.